cpu/stm32f3/periph/pwm.c

```diff
@@ -134,37 +142,44 @@ int pwm_set(pwm_t dev, int channel, unsigned int value)
 
     switch (dev) {
 #if PWM_0_EN
+
         case PWM_0:
             tim = PWM_0_DEV;
             break;
 #endif
 #if PWM_1_EN
+
         case PWM_1:
             tim = PWM_1_DEV;
             break;
 #endif
+
         default:
             return -1;
     }
 
-    /* norm value to maximum possible value */
-    if (value > 0xffff) {
-        value = 0xffff;
+    /* norm value to actual resolution */
+    if (value > tim->ARR) {
+        value = tim->ARR;
```

Comment:
I followed the hint from @PeterKietzmann.
> the example in _fig. 174_ on page 537-538 please? Does it answer the question about the clipping of the duty cycle? If so, I'd suggest to stay with the `0xffff` solution and add a comment. 

Using values `> tim->ARR` seems to safe on this MCU. Even though, why not doing it generally correct instead of placing a comment pointing to possible problems? Aside from one pointer dereference, reading a register isn't more expensive as reading a constant from RAM, in a memory mapped system. *My 5ct.*

If @haukepetersen and @PeterKietzmann stick to their veto I will skip this commit.

> correctly I'd vote for your solution but to add a comment here. Otherwise, of course

Would you agree with this comment replacement?

    /* norm value to maximum possible value.
     * Caution! A duty cycle overriding the period lenth can lead to unexpected
     * behaviour on other MCUs. */